Argo/Globals/Parser.swift

```diff
@@ -0,0 +1,26 @@
+import Box
+
+public enum Parser<T> {
+  case Success(Box<T>)
+  case TypeMismatch(String)
+  case MissingKey(String)
+
+  public var value: T? {
+    switch self {
+    case let .Success(box): return box.value
+    default: return .None
+    }
+  }
+
+  public static func typeMismatch(expectedType: String, object: JSON) -> Parser<T> {
+    return .TypeMismatch("\(object) is not a \(expectedType)")
+  }
+}
+
+func parseOptional<A>(p: Parser<A>) -> Parser<A?> {
```

Comment:
Personally, I'd rather `optional` since I don't get that this is doing any kind of parsing.